node/bft/src/helpers/proposal.rs

```diff
@@ -167,6 +168,86 @@ impl<N: Network> Proposal<N> {
     }
 }
 
+impl<N: Network> ToBytes for Proposal<N> {
+    fn write_le<W: Write>(&self, mut writer: W) -> IoResult<()> {
+        // Write the batch header.
+        self.batch_header.write_le(&mut writer)?;
+        // Write the number of transmissions.
+        u32::try_from(self.transmissions.len()).map_err(error)?.write_le(&mut writer)?;
+        // Write the transmissions.
+        for (transmission_id, transmission) in &self.transmissions {
+            transmission_id.write_le(&mut writer)?;
+            transmission.write_le(&mut writer)?;
+        }
+        // Write the number of signatures.
+        u32::try_from(self.signatures.len()).map_err(error)?.write_le(&mut writer)?;
+        // Write the signatures.
+        for signature in &self.signatures {
+            signature.write_le(&mut writer)?;
+        }
+        Ok(())
+    }
+}
+
+impl<N: Network> FromBytes for Proposal<N> {
+    fn read_le<R: Read>(mut reader: R) -> IoResult<Self> {
+        // Read the batch header.
+        let batch_header = FromBytes::read_le(&mut reader)?;
+        // Read the number of transmissions.
+        let num_transmissions = u32::read_le(&mut reader)?;
```

Comment:
should we impose a limit on the number of transmissions (same goes for signatures), or do we only ever expect to create and load them locally?

Reply:
Resolved with [c53f5fd](https://github.com/AleoHQ/snarkOS/pull/3239/commits/c53f5fdc50cc47c68b04dad81e380e339e708e7b)